Backend/ActiveCruzer/BLL/MyRequestBll.cs

```diff
@@ -0,0 +1,159 @@
+        /// <param name="requestId"></param>
+        /// <param name="userId"></param>
+        public void AbortRequest(int requestId, int userId)
+        {
+            var request = _context.Request.FirstOrDefault(x => x.Id == requestId);
+            if(request != null)
+            {
+                request.Status = Request.RequestStatus.Open;
+                request.Volunteer = null;
+                _context.Update(request);
+                _context.SaveChanges();
+            }
+        }
+
+        /// <summary>
+        /// return request with given request id
+        /// </summary>
+        /// <param name="requestId"></param>
+        /// <param name="userId"></param>
+        /// <returns></returns>
+        public Request GetRequest(int requestId)
+        {
+            var request = _context.Request.FirstOrDefault(x => x.Id == requestId);
+            if(request != null)
+            {
+                return request;
+            }
+            return null;
+        }
+
+        /// <summary>
+        /// dispose from inherit
+        /// </summary>
+        public void Dispose()
+        {
+
+        }
+
+        /// <summary>
+        /// check if request exsists on user with param user id and request id
+        /// </summary>
+        /// <param name="requestId"></param>
+        /// <param name="userId"></param>
+        /// <returns></returns>
+        public bool ExistsOnUser(int requestId, int userId)
+        {
+            var request = _context.Request.FirstOrDefault(x => x.Id == requestId);
+            if(request != null)
+            {
+                return request.Id == requestId && request.Volunteer == userId;
+            }
+            return false;
+        }
+
+        /// <summary>
+        /// get all reqeusts of user based on hardcoded user
+        /// </summary>
+        /// <param name="hardcodedUser"></param>
+        /// <returns></returns>
+        public List<Request> GetAllPendingFromUser(int hardcodedUser)
+        {
+            return _context.Request.Where(it => it.Volunteer == hardcodedUser && it.Status == Request.RequestStatus.Pending).ToList();
+        }
+
+        /// <summary>
+        /// does requests exist bool?
+        /// </summary>
+        /// <param name="requestId"></param>
+        /// <returns></returns>
+        public bool Exists(int requestId)
+        {
+            var request = _context.Request.FirstOrDefault(x => x.Id == requestId);
+            if(request != null)
+            {
+                return true;
+            }
+            return false;
+        }
+
+        /// <summary>
+        /// reuquest not closed bool?
+        /// </summary>
+        /// <param name="requestId"></param>
+        /// <returns></returns>
+        public bool IsNotClosed(int requestId)
+        {
+            var request = _context.Request.FirstOrDefault(x => x.Id == requestId);
+            if(request != null)
+            {
+                return true;
```

Comment:
I think that we miss a check if the request is open

Reply:
implemented check